app/src/main/java/com/kms/cura/view/CheckBoxAdapter.java

```diff
@@ -0,0 +1,98 @@
+package com.kms.cura.view;
+
+import android.content.Context;
+import android.database.DataSetObserver;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.BaseAdapter;
+import android.widget.CheckBox;
+import android.widget.LinearLayout;
+import android.widget.SpinnerAdapter;
+import android.widget.TextView;
+
+import com.kms.cura.R;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by toanbnguyen on 5/30/2016.
+ */
+public class CheckBoxAdapter extends BaseAdapter {
+    Context context;
+    int resource;
+    ArrayList<String> checkList;
+    boolean[] checked;
+    public CheckBoxAdapter(Context context, int resource, ArrayList objects, boolean[] checked) {
```

Comment:
ArrayList of built-in type cannot be put into a bundle for transferring data between fragments, so I use normal array for boolean to avoid extra works (Bundle class supports to put in that array). Instead I suppose making this private too.